cinema/views.py

```diff
@@ -24,21 +28,55 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed")
```

Comment:
Raising a `NotFound` exception for unsupported actions is not appropriate. Instead, consider using `MethodNotAllowed` to indicate that the HTTP method is not allowed for the requested resource.